storage/backup/backup-cli/src/coordinators/backup.rs

```diff
@@ -397,18 +445,20 @@ impl BackupCompactor {
         }
 
         // Move the compacted metadata files to the metadata backup folder
-        for file in files {
+        let to_move = self.update_existing_file_timestamps(&mut metaview, files, new_files)?;
+        for file in to_move {
             // directly return if any of the backup task fails
             info!(file = file, "Backup metadata file.");
-            let name = Path::new(&file)
-                .file_name()
-                .and_then(OsStr::to_str)
-                .ok_or_else(|| format_err!("cannot extract filename from {}", file))?;
-            if !new_files.contains(name) {
-                self.storage.backup_metadata_file(&file).await?
-            }
+            self.storage.backup_metadata_file(&file).await?
         }
 
+        // save the metadata compaction timestamps
+        self.storage
+            .save_metadata_line(
+                &Metadata::get_compaction_timestamps_file_name().try_into()?,
```

Comment:
We are assuming files can be peacefully rewritten here, which I think might be not necessary. How about each time use a different name (maybe append the timestamp), and "compact" these files as well? 
Add a timestamp in the metadata, so when loading we can dedup properly (keeping only the latest).

Reply:
Yes, I am thinking of the impact of the failed update. If this line failed or is delayed to update the timestamps, the next compaction run would still read all the metadata and regenerate the same compacted files with a current timestamp.  For original metadata files, either no impact or keep them longer before eventually deleting them.

Reply:
no overwrite now. each file has a timestamp